modules/core/03-connection/types/msgs.go

```diff
@@ -102,6 +106,10 @@ func NewMsgConnectionOpenTry(
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgConnectionOpenTry) ValidateBasic() error {
+	if msg.ClientId == exported.Localhost {
+		return sdkerrors.Wrap(clienttypes.ErrInvalidClientType, "localhost connetion handshakes are disallowed")
+	}
```

Comment:
I think this isn't strictly necessary as there should not exist a connection in INIT which would have a valid consensus state stored for the localhost. But, I think it makes sense to be clear/defensive

Reply:
Yeah, exactly! This was my thinking too :)